packages/mui-base/src/Modal/Modal.tsx

```diff
@@ -405,6 +405,18 @@ Modal.propTypes /* remove-proptypes */ = {
    * @param {string} reason Can be: `"escapeKeyDown"`, `"backdropClick"`.
    */
   onClose: PropTypes.func,
+  /**
+   * Callback function called when a transition enters.
+   *
+   * @returns {void}
+   */
```

Comment:
We don't generally document types here as this is handled by TypeScript.

Also, "**callback** function **called**" sounds repetitive. 

```suggestion
  /**
   * A function called when a transition enters.
   */
```

Please update the other occurrences as well.